shared_model/validators/block_validator.hpp

```diff
@@ -32,18 +34,27 @@ namespace shared_model {
     /**
      * Class that validates block
      */
-    class BlockValidator {
+    template <typename FieldValidator, typename TransactionValidator>
+    class BlockValidator
```

Comment:
It is not clear why BlockValidator and ProposalValidators inherit from ContainerValidator, because they do not override any behaviour defined in ContainerValidator. It seems that `SignableValidator` is a relatively good example for inheritance, because it extends the behaviour of base validator. Consider making a template class, which define base validation for transactions, height, and created time, and extending this class in BlockValidator and ProposalValidator.